delete files only matching regexp

tarasglek opened the pull request:
I like this script, but would like to delete video files in my securitycam archive and keep pics. This achieves that.

I can now prioritize deletion of videos over static pics by issuing command in sequence
```
free-disk --free-bytes 400GB -d --delete-re '.*mp4$' test_dir/
free-disk --free-bytes 400GB -d --delete-re '.*jpg$' test_dir/
```

fphammerle commented:
Thanks! I'll merge it after I added a test case (might take a few days)

fphammerle commented:
Thanks! Could you rebase?

tarasglek commented:
Hey, rebased on dev. I realized this still wasn't working right on ZFS added a workaround. Without workaround, i'd ask free-disk to free 1gb, and it would free 12gb.

fphammerle commented:
Thanks! I merged the regex part renaming the param to `--delete-path-regex` (to be more explicit) and replacing `re.match` with `re.search` cause I thought it was more intuitive not to imply matching at start of path (rather specify `^` explicitly). released as v1.0.0

Before merging the second part (disk usage bug fix), I'll need to add tests first

tarasglek commented:
For more features see https://github.com/tarasglek/free-disk/tree/taras/precision.

I added --dry-run and units to debug output to help me debug. Also changed exit code to make chaining more robust.
